### PR TITLE
Jenkins: load and test HPX support on daint

### PR DESCRIPTION
**Pull request by havogt**

### CODE REVIEW

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
Yes, from within the ETH network. As expected from the slack comments, the debug build with release hpx failed.

File: jenkins/envs/daint_nvcc_gcc.sh [hunk 1]
```diff
@@ -3,7 +3,10 @@
 source $(dirname "$BASH_SOURCE")/daint.sh
 
 module swap PrgEnv-cray PrgEnv-gnu
-module swap gcc/7.3.0
+
+if [ "$build_type" == "release" ]; then
```
Comment (havogt) [inline jenkins/envs/daint_nvcc_gcc.sh:7]: HPX release build (the version installed on daint) is not compatible with debug builds.
Do you have a better idea to solve this @fthaler 
Problem is, if you load the env and configure manually you will not have HPX in your env and behavior is different than on jenkins...

**Comment by havogt** [comment]
launch jenkins

File: jenkins/envs/daint_nvcc_gcc.sh (hunk 1 above)
Comment (fthaler) [reply jenkins/envs/daint_nvcc_gcc.sh:7]: Hmm not really. Maybe setting the condition to `!= "debug"` helps a bit? Then it loads the module by default and the user should see a message if configuration fails and thus maybe checks the env file. But I feel like there should be either a debug build available on Daint or they should allow this, right?

Comment (havogt) [reply jenkins/envs/daint_nvcc_gcc.sh]: Thanks makes sense.
Yes, they are aware that it's not a good pattern...

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
tests passed

File: jenkins/envs/daint_nvcc_gcc.sh (hunk 1 above)
Comment (msimberg) [reply jenkins/envs/daint_nvcc_gcc.sh]: In case you want to track our progress on fixing that: https://github.com/STEllAR-GROUP/hpx/issues/5013.

**Pull request merged**